pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -182,36 +198,45 @@ func (am *Alertmanager) checkReadiness(ctx context.Context) error {
 // CompareAndSendConfiguration checks whether a given configuration is being used by the remote Alertmanager.
 // If not, it sends the configuration to the remote Alertmanager.
 func (am *Alertmanager) CompareAndSendConfiguration(ctx context.Context, config *models.AlertConfiguration) error {
-	c, err := notifier.Load([]byte(config.AlertmanagerConfiguration))
+	// Decrypt the configuration before comparing.
+	rawDecrypted, err := am.decryptConfiguration(ctx, config.AlertmanagerConfiguration)
 	if err != nil {
 		return err
 	}
 
-	// Decrypt the configuration before comparing.
+	// Send the configuration only if we need to.
+	if !am.shouldSendConfig(ctx, rawDecrypted) {
+		return nil
+	}
+
+	return am.sendConfiguration(ctx, string(rawDecrypted), config.ConfigurationHash, config.CreatedAt, config.Default)
+}
```

Comment:
This method works in the same way as before, I just extracted some lines to `am.decryptConfiguration()` and `am.sendConfiguration()`.